daemons/attrd/attrd_ipc.c

```diff
@@ -596,7 +596,7 @@ attrd_ipc_dispatch(qb_ipcs_connection_t * c, void *data, size_t size)
 
     if (xml == NULL) {
         crm_debug("Unrecognizable IPC data from PID %d", pcmk__client_pid(c));
-        pcmk__ipc_send_ack(client, id, flags, PCMK__XE_ACK, NULL,
+        pcmk__ipc_send_ack(client, id, flags, PCMK__XE_NACK, NULL,
```

Comment:
I _think_, but am not remotely sure, that some clients may need to be updated -- either to return early, or to properly handle a NACK in some other way.

Previously, the only things that sent NACK were the CIB manager (for invalid message), the fencer, and the executor (in `remote_proxy_cb()`).

No client looks for NACK, but they do look for ACK. The following clients' IPC dispatch functions return immediately if they receive an ACK (but not a NACK):
* attrd
* controld
* pacemakerd
* schedulerd

---

I'm starting to think it'd be simpler to drop `PCMK__XE_NACK` and drop the `tag` argument of `pcmk__ipc_send_ack()`, and instead send `PCMK__XE_ACK` unconditionally. Clients can use the exit code.

Reply:
Yeah, that's kind of what I was thinking too based on the commit message.  I don't remember why we have both ACK vs. NACK and an exit code.  One other thing to note here is that not every message gets an ACK, either.  I'm also not convinced that any of this matters too terribly much either given that we control both sides of the connection and it takes place on the same system with what I assume will always be the same versions of the pacemaker daemons.

I'd be fine with dropping this patch, merging this PR, and then dealing with the ACK vs. NACK stuff in a later PR.